firmware/common/portapack_persistent_memory.cpp

```diff
@@ -923,5 +931,122 @@ size_t data_size() {
+        pmem_dump_file.write_line("ui_config show_bigger_qr_code: " + to_string_dec_uint(data->ui_config.show_bigger_qr_code()));
+        pmem_dump_file.write_line("ui_config disable_touchscreen: " + to_string_dec_uint(data->ui_config.disable_touchscreen()));
+        pmem_dump_file.write_line("ui_config hide_clock: " + to_string_dec_uint(data->ui_config.hide_clock()));
+        pmem_dump_file.write_line("ui_config clock_with_date: " + to_string_dec_uint(data->ui_config.clock_with_date()));
+        pmem_dump_file.write_line("ui_config clkout_enabled: " + to_string_dec_uint(data->ui_config.clkout_enabled()));
+        pmem_dump_file.write_line("ui_config stealth_mode: " + to_string_dec_uint(data->ui_config.stealth_mode()));
+        pmem_dump_file.write_line("ui_config config_login: " + to_string_dec_uint(data->ui_config.config_login()));
+        pmem_dump_file.write_line("ui_config config_splash: " + to_string_dec_uint(data->ui_config.config_splash()));
+        // misc_config bits
+        pmem_dump_file.write_line("misc_config config_audio_mute: " + to_string_dec_int(config_audio_mute()));
+        pmem_dump_file.write_line("misc_config config_speaker_disable: " + to_string_dec_int(config_speaker_disable()));
+
+        // receiver_model
+        pmem_dump_file.write_line("[Receiver Model]");
+        pmem_dump_file.write_line("target_frequency: " + to_string_dec_uint(receiver_model.target_frequency()));
+        pmem_dump_file.write_line("frequency_step: " + to_string_dec_uint(receiver_model.frequency_step()));
+        pmem_dump_file.write_line("lna: " + to_string_dec_int(receiver_model.lna()));
+        pmem_dump_file.write_line("vga: " + to_string_dec_int(receiver_model.vga()));
+        pmem_dump_file.write_line("rf_amp: " + to_string_dec_int(receiver_model.rf_amp()));
+        pmem_dump_file.write_line("baseband_bandwidth: " + to_string_dec_uint(receiver_model.baseband_bandwidth()));
+        pmem_dump_file.write_line("sampling_rate: " + to_string_dec_uint(receiver_model.sampling_rate()));
+        switch (receiver_model.modulation()) {
+            case ReceiverModel::Mode::AMAudio:
+                pmem_dump_file.write_line("modulation: Mode::AMAudio");
+                break;
+            case ReceiverModel::Mode::NarrowbandFMAudio:
+                pmem_dump_file.write_line("modulation: Mode::NarrowbandFMAudio");
+                break;
+            case ReceiverModel::Mode::WidebandFMAudio:
+                pmem_dump_file.write_line("modulation: Mode::WidebandFMAudio");
+                break;
+            case ReceiverModel::Mode::SpectrumAnalysis:
+                pmem_dump_file.write_line("modulation: Mode::SpectrumAnalysis");
+                break;
+            case ReceiverModel::Mode::Capture:
+                pmem_dump_file.write_line("modulation: Mode::Capture");
+                break;
+        }
```

Comment:
Compilers complains that default  is unused so I removed the default, I guess it's because I'm passing it a 'Mode' and it has defined states

Reply:
Ho well, it do not complain anymore. added !

Reply:
Yeah, but you can always do this: `auto m = (Mode)50;` which is what I would worry about if there was junk in the memory. :(
That's kind of a wonky compiler warning because it's just wrong actually.

Reply:
I guess you could tell by the fact that line would be missing from the output, but that's kind of lame. What the warning number?

Reply:
Maybe print a line with raw value above the switch then or build a string?
auto mod_str = to_string...(modulation);
switch (modulation)
  case a: mod_str += " AM";
 
 write_line("mod: " + mod_str);
 ```
 
 Something along those lines?